src/insert.js

```diff
@@ -31,7 +31,7 @@ const insert: Mutator<any> = (
         if (fieldIndex >= index) {
           // inc index one higher
           const incrementedKey = `${name}[${fieldIndex + 1}]${tokens[2]}`
-          moveFieldState(state, backup[key], incrementedKey)
+          moveFieldState(state, backup.fields[key], incrementedKey, backup)
```

Comment:
@SeqviriouM Found your PR while was trying to resolve issues with `insert`.
Tested this code in app, where I have issues with `insert` and it didn't work well. Deeply nested fields handlers are messed up, so changing value in one field changes value in another one(I think because of moving handlers). What helped me is just:
```
delete state.fields[key]
```
So all fields after inserted one get default state and handlers.